src/uproot/interpretation/objects.py

```diff
@@ -205,7 +205,11 @@ def final_array(
 
             start = stop
 
-        if all(uproot._util.from_module(x, "awkward") for x in basket_arrays.values()):
+        if len(basket_arrays) == 0:
+            output = numpy.array([], dtype=self.numpy_dtype)
+        elif all(
+            uproot._util.from_module(x, "awkward") for x in basket_arrays.values()
+        ):
```

Comment:
This was an actual bug-fix, discovered while arranging these tests but otherwise unrelated to them.